core/src/com/unciv/logic/map/AncientRuins.kt

```diff
@@ -0,0 +1,147 @@
+package com.unciv.logic.map
+
+import com.badlogic.gdx.graphics.Color
+import com.unciv.Constants
+import com.unciv.UncivGame
+import com.unciv.logic.civilization.CivilizationInfo
+import com.unciv.models.ruleset.tech.TechEra
+import com.unciv.models.stats.Stat
+import com.unciv.ui.pickerscreens.RuinBonusPickerScreen
+import com.unciv.ui.worldscreen.unit.UnitActions
+import kotlin.random.Random
+
+class AncientRuins {
+
+    val civInfo: CivilizationInfo
+    val tile: TileInfo
+    val unit: MapUnit
+    val actions: ArrayList<RuinAction>
+    val randomSeed: Random
+
+    companion object {
+        private const val ANCIENT_RUIN_MAP_REVEAL_OFFSET = 4
+        private const val ANCIENT_RUIN_MAP_REVEAL_RANGE = 4
+        private const val ANCIENT_RUIN_MAP_REVEAL_CHANCE = 0.8f
+    }
+
+    constructor(civInfo: CivilizationInfo, tile: TileInfo, unit: MapUnit) {
+        this.civInfo = civInfo
+        this.tile = tile
+        this.unit = unit
+
+        tile.improvement=null
+        randomSeed = Random(tile.position.toString().hashCode())
+        actions = ArrayList()
+        var blockedActions = civInfo.updateAndGetBlockedRuinsBonus()
+        if(civInfo.cities.isNotEmpty() && !blockedActions.contains(RuinBonus.POPULATION)) {
```

Comment:
This function should be cleaner of you filtered according to blocked functions at the end, instead of per evety sibgle available value - there are a lot of 'if's here, and quite a bit of nesting